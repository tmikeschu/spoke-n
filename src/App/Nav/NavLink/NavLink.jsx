import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router'

const NavLink = props => (
  <Link {...props} activeClassName="active" />
)

NavLink.propTypes = {
  to: PropTypes.string.isRequired
}

export default NavLink
