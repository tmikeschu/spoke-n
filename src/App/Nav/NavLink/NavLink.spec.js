import React from 'react'
import { shallow } from "enzyme"
import NavLink from "./NavLink"
import { Link } from 'react-router'

describe("<NavLink />", () => {
  const navLink = shallow(<NavLink to="/" />)

  it("renders without crashing", () => {
    expect(navLink).toBeTruthy()
  })

  it("renders the correct link", () => {
    expect(navLink.equals(
      <Link to="/" activeClassName="active" />
    )).toBeTruthy()
  })

  describe("snapshot", () => {
    it("is valid", () => {
      expect(navLink.getNodes()).toMatchSnapshot()
    })
  })
})
