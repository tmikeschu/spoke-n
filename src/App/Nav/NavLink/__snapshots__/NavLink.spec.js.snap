// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLink /> snapshot is valid 1`] = `
Array [
  <Link
    activeClassName="active"
    onlyActiveOnIndex={false}
    style={Object {}}
    to="/"
  />,
]
`;
